### ISSUE TITLE
Improve error message when missing phpunit.xml in 3.x and 4.x

### ISSUE THREAD

**Issue by SilbinaryWolf** [silverstripe/silverstripe-framework#7229]
Getting the following messages when you're missing phpunit.xml isn't helpful and is confusing to new users or those new to writing unit tests:
`Fatal error: Class 'SapphireTest' not found`
`Fatal error: Class 'FunctionalTest' not found`

Is it possible to add some sort of configuration or hook to improve this?
Perhaps document the issue at least?

An example call stack is:
```
Fatal error: Class 'FunctionalTest' not found in /var/www/dynamic/modules/test-assist/test-assist/tests/SiteTreeCMSFieldsTest.php on line 7

Call Stack:
    0.0003     224080   1. {main}() /var/www/dynamic/modules/test-assist/vendor/phpunit/phpunit/composer/bin/phpunit:0
    0.0121     541424   2. PHPUnit_TextUI_Command::main() /var/www/dynamic/modules/test-assist/vendor/phpunit/phpunit/composer/bin/phpunit:63
    0.0122     542048   3. PHPUnit_TextUI_Command->run() /var/www/dynamic/modules/test-assist/vendor/phpunit/phpunit/PHPUnit/TextUI/Command.php:129
    0.0211     789360   4. PHPUnit_Runner_BaseTestRunner->getTest() /var/www/dynamic/modules/test-assist/vendor/phpunit/phpunit/PHPUnit/TextUI/Command.php:150
    0.0295     985984   5. PHPUnit_Framework_TestSuite->addTestFiles() /var/www/dynamic/modules/test-assist/vendor/phpunit/phpunit/PHPUnit/Runner/BaseTestRunner.php:96
    0.0295     986808   6. PHPUnit_Framework_TestSuite->addTestFile() /var/www/dynamic/modules/test-assist/vendor/phpunit/phpunit/PHPUnit/Framework/TestSuite.php:416
    0.0317    1072584   7. PHPUnit_Util_Fileloader::checkAndLoad() /var/www/dynamic/modules/test-assist/vendor/phpunit/phpunit/PHPUnit/Framework/TestSuite.php:355
    0.0317    1072792   8. PHPUnit_Util_Fileloader::load() /var/www/dynamic/modules/test-assist/vendor/phpunit/phpunit/PHPUnit/Util/Fileloader.php:76
    0.0322    1079432   9. include_once('/var/www/dynamic/modules/test-assist/test-assist/tests/SiteTreeCMSFieldsTest.php') /var/www/dynamic/modules/test-assist/vendor/phpunit/phpunit/PHPUnit/Util/Fileloader.php:92
```

**Comment by dhensby** [comment]
If you're running `vendor/bin/phpunit` then this is all phpunit land so there's not much we can bootstrap into the process because that's defined by the file which is missing...

**Comment by SilbinaryWolf** [comment]
Yeah that was what I assumed from the callstack.

Maybe a yellow warning block on this page:
https://docs.silverstripe.org/en/4/developer_guides/testing/unit_testing/

Saying "If your phpunit.xml file is missing, you will get `Fatal error: Class 'SapphireTest' not found` or `Fatal error: Class 'FunctionalTest' not found`". This will help people googling the problem to discover the solution with ease.

**Comment by dhensby** [comment]
Sure, would you like to PR @SilbinaryWolf?

**Comment by SilbinaryWolf** [comment]
No thank you!

**Comment by tractorcow** [comment]
In 4.x the class shouldn't be un-findable, since composer psr-4 will ensure it's findable out of the box. We WILL get "class not found" if phpunit isn't installed, however.

In 3.x the most we should need to do is update the docs as suggested.

**Comment by dhensby** [comment]
PHPUnit doesn't automatically include the composer autoloader - it has to be included via the bootstrap (the autoloader can be the bootstrap).

**Comment by tractorcow** [comment]
Yes it does see `vendor/bin/phpunit`. That's how phpunit loads it's own classes.

**Comment by dhensby** [comment]
right, I stand corrected, PHPUnit has been including the composer autoloader for a long time.

fix to this issue at #7514

**Comment by tractorcow** [comment]
Fixed with #7514

**Comment by SilbinaryWolf** [comment]
Awesome work both of you. Thanks a lot!
